tests/core/test_config.py

```diff
@@ -203,3 +204,35 @@ class Z:
     result = from_dict(Z, data, Config(strict_unions_match=True))
 
     assert result == Z(u=Y(f=1))
+
+
+def test_custom_from_dict_in_nested_data_class():
+    @dataclass
+    class X:
+        d: date
+        t: str
+
+        def from_dict(data_class, data, config):
+            data["t"] = "prefix {}".format(data["t"])
```

Comment:
This prefix can be done with the builtin `__post_init__` too

Reply:
Yes, but this is just an example. The main goal is to have a place to reshape the dictionary before we create a data class from it.